lms/djangoapps/bulk_email/tasks.py

```diff
@@ -522,8 +556,26 @@ def _send_course_email(entry_id, email_id, to_list, global_email_context, subtas
             # Pop the user that was emailed off the end of the list only once they have
             # successfully been processed.  (That way, if there were a failure that
             # needed to be retried, the user is still on the list.)
+            try:
```

Comment:
if `recipients_info` is a counter, you won't get the `KeyError` (that was the whole point of using one)